manager/integration/tests/test_orphan.py

```diff
@@ -0,0 +1,940 @@
+from common import exec_nsenter
+from common import get_self_host_id
+from common import get_volume_endpoint
+from common import get_update_disks, wait_for_disk_update, cleanup_node_disks
+from common import create_and_check_volume, wait_for_volume_healthy
+from common import generate_random_id
+from common import wait_for_orphan_count, delete_orphan
+
+
+def create_host_disk(client, vol_name, size, node_id):  # NOQA
+    # create a single replica volume and attach it to node
+    volume = create_volume(client, vol_name, size, node_id, 1)
+
+    mkfs_ext4_settings = client.by_id_setting(SETTING_MKFS_EXT4_PARAMS)
+    mkfs_ext4_options = mkfs_ext4_settings.value
+
+    # prepare the disk in the host filesystem
+    disk_path = common.prepare_host_disk(get_volume_endpoint(volume),
+                                         volume.name,
+                                         mkfs_ext4_options)
+    return disk_path
+
+
+def cleanup_host_disk(client, *args):  # NOQA
+    # clean disk
+    for vol_name in args:
+        # umount disk
+        common.cleanup_host_disk(vol_name)
+        # clean volume
+        cleanup_volume(client, vol_name)
+
+
+def create_volume(client, vol_name, size, node_id, r_num):  # NOQA
+    volume = client.create_volume(name=vol_name, size=size,
+                                  numberOfReplicas=r_num)
+    assert volume.numberOfReplicas == r_num
+    assert volume.frontend == VOLUME_FRONTEND_BLOCKDEV
+
+    volume = common.wait_for_volume_detached(client, vol_name)
+    assert len(volume.replicas) == r_num
+
+    assert volume.state == "detached"
+    assert volume.created != ""
+
+    volumeByName = client.by_id_volume(vol_name)
+    assert volumeByName.name == volume.name
+    assert volumeByName.size == volume.size
+    assert volumeByName.numberOfReplicas == volume.numberOfReplicas
+    assert volumeByName.state == volume.state
+    assert volumeByName.created == volume.created
+
+    volume.attach(hostId=node_id)
+    volume = wait_for_volume_healthy(client, vol_name)
+
+    return volume
+
+
+def cleanup_volume(client, vol_name):  # NOQA
+    volume = client.by_id_volume(vol_name)
+    volume.detach(hostId="")
+    client.delete(volume)
+    common.wait_for_volume_delete(client, vol_name)
+
+
+@pytest.mark.orphan
+def test_orphaned_dirs_with_wrong_naming_format(client, volume_name):  # NOQA
+    """
+    Test orphan CRs are not created for the orphaned directories with wrong
+    naming formats
+    1. Create a new disk holding valid and invalid orphaned replica
+       directories
+    2. Create a volume and then attach to the current node
+    3. Create one valid orphaned replica directories by copying the active
+       replica directory
+    4. Create multiple invalid orphan replica directories with wrong naming
+       format
+    5. Verify orphan list only contains the orphan CR for valid orphaned
+       replica directory
+    """
+
+    lht_hostId = get_self_host_id()
```

Comment:
Because we don't have BBD test case atm, please add a comment for each step at the beginning of it like "# Step 1" and it will be helpful for reading/development/debugging.

Reply:
Addressed!